testing/test_unittest.py

```diff
@@ -192,30 +191,32 @@ def test_check(self):
 def test_teardown_issue1649(pytester: Pytester) -> None:
```

Comment:
The way this test was written previously wasn't good; it checked if the obj is still alive *after* the pytester run, at which the entire session was torn down, too late. This fixes it up. I made sure it now fails before and passes after.